lib/include/cuttlefish/Texture.h

```diff
@@ -611,6 +622,13 @@ class CUTTLEFISH_EXPORT Texture
 	 */
 	SaveResult save(const char* fileName, FileType fileType = FileType::Auto);
 
+    /**
+     * @brief Saves a texture to a std::vector<char>.
+     * @param out A std::vector<char> to store the data to. This buffer will be resized and overwritten.
+     * @param fileType The type of the file to save.
+     * @return False if the texture hasn't been converted or the data couldn't be written.
+     */
+    SaveResult save(std::vector<unsigned char>& out, FileType fileType);
```

Comment:
When I address any cleanup after this is merged, I plan on changing this to std::uint8_t. It should still be compatible with unsigned char, but I wanted to give you a heads up in case you see that and worry that it might break your calling code.

Reply:
Actually now that I have refactored this code to use ostream internally I could change this to a template function to support any "8 bit sized" vector.. (though how do I FORCE it to be that size!)

Reply:
Though in the end it shouldn't make much difference at all.. (and uint8_t is probably the better choice here anyways).

I can switch it to uint8_t now if you wish.